src/manager.cpp

```diff
@@ -93,6 +93,121 @@ uint32_t SessionManager::closeAllByType(SessionType type)
     return count;
 }
 
+uint32_t SessionManager::closeUserSessionsByType(SessionType type,
+                                                 std::string sessionId,
+                                                 bool ownedOnly)
+{
+    auto numSessId = parseSessionId(sessionId);
+    std::string userName = getSessionItem(sessionId)->getOwner();
+    bool allowed = isAllSessionsAllowed(sessionId);
+    const auto count =
+        std::erase_if(sessionItems, [allowed, ownedOnly, type, userName,
+                                     numSessId](const auto& item) {
+            auto const [id, session] = item;
+            if ((allowed) && !(ownedOnly))
+            {
+                return session->sessionType() == type && id != numSessId;
+            }
+            else
+            {
+                return session->sessionType() == type &&
+                       session->getOwner() == userName && id != numSessId;
+            }
+        });
+    return count;
+}
+
+bool SessionManager::isAllSessionsAllowed(std::string sessionId)
+{
+    log<level::DEBUG>("SessionManager::isOwnSession()",
+                      entry("CALLER-SESSION-ID=%d", sessionId.c_str()));
```

Comment:
Use `%s` for string data.

Reply:
@nest1ing Fixed + eveyrwhere in my changes I found the same issue also fixed
there are also similar issues with legacy code, but I am afraid I am not brave enough to change legacy code without permission )